activejob/lib/active_job/log_subscriber.rb

```diff
@@ -77,6 +77,8 @@ def queue_name(event)
       end
 
       def args_info(job)
+        return "" if job.arguments_logging_disabled?
+
         if job.arguments.any?
```

Comment:
What about flipping to `job.log_arguments?` and use it as `if job.log_arguments? && job.arguments.any?`?